src/components/players/list/card.js

```diff
@@ -0,0 +1,95 @@
+import React from 'react';
+import { withRouter } from 'react-router-dom';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+
+import placeholderImg from 'assets/imgs/placeholder.png';
+import { formatPlayerPhotoUrl } from 'utils/stringUtils';
+
+import styles from './index.module.scss';
+
+const Card = ({
+  getSelectedPlayer,
+  history,
+  player,
+  player: { first_name, height_ft, height_in, jersey_number, last_name, person_id, position_full, weight_lbs },
+  selectedPlayerId,
+  selectedTeam: { teamColor, teamId, urlName },
+}) => {
+  const isSelected = person_id === selectedPlayerId;
+  return (
+    <div
+      onClick={() => {
+        history.push(`/${urlName}/players/${person_id}`);
+        getSelectedPlayer(player);
+      }}
+      className={classnames(styles.playerCard, isSelected && styles.selectedCard)}>
+      <div className={styles.imageContainer}>
+        <img
+          src={person_id ? formatPlayerPhotoUrl(teamId, person_id) : placeholderImg}
+          alt="player headshot"
+          onError={(e) => (e.target.src = placeholderImg)}
+        />
+      </div>
+      <div style={{ borderColor: teamColor }} className={styles.imageLine} />
+      <div style={{ backgroundColor: isSelected && teamColor }} className={styles.detailsContainer}>
+        {!!Object.keys(player).length && (
+          <>
+            <div className={styles.number}>{jersey_number}</div>
+            <div className={styles.details}>
+              <div className={styles.name}>
+                {first_name} {last_name}
+              </div>
+              <div className={styles.position}>{position_full.replace('-', ' - ')}</div>
+              <div className={styles.size}>
+                {height_ft}' {height_in}, {weight_lbs} lbs
+              </div>
+            </div>
+          </>
+        )}
+      </div>
+    </div>
+  );
+};
+
+Card.propTypes = {
+  formatPlayerPhotoUrl: PropTypes.func,
+  getSelectedPlayer: PropTypes.func.isRequired,
+  history: PropTypes.object,
+  player: PropTypes.object.isRequired,
+  placeholderImg: PropTypes.string,
+  first_name: PropTypes.string,
+  height_ft: PropTypes.string,
+  height_in: PropTypes.string,
+  jersey_number: PropTypes.string,
+  last_name: PropTypes.string,
+  person_id: PropTypes.string,
+  position_full: PropTypes.string,
+  selectedPlayerId: PropTypes.string,
+  teamColor: PropTypes.string,
+  teamId: PropTypes.string,
+  urlName: PropTypes.string,
+  weight_lbs: PropTypes.string,
+  isSelected: PropTypes.bool,
```

Comment:
Too many props (also they aren't declared correctly). It should be:

```
Card.propTypes = {
  getSelectedPlayer: PropTypes.func.isRequired,
  selectedTeam: PropTypes.object.isRequired,
  player: PropTypes.object.isRequired,
  selectedPlayerId: PropTypes.string,
};

Card.defaultProps = {
  selectedPlayerId: null,
};
```

---
An aside, remember that the other props you are declaring such as `first_name`, `height_ft` are subprops and would be declared using `Proptypes.shape`:

```
player: Proptypes.shape({
  first_name: PropTypes.string,
  height_ft: PropTypes.string,
  height_in: PropTypes.string,
  ...
}).isRequired
```

But as we stated, no need to declare subprops in cases like this.

Reply:
True, forgot about shape.